core/src/main/java/com/gdx/game/screen/GameScreen.java

```diff
@@ -92,6 +99,31 @@ public GameScreen(GdxGame gdxGame, ResourceManager resourceManager) {
         player = EntityFactory.getInstance().getEntity(ProfileManager.getInstance().getProperty("playerCharacter", EntityFactory.EntityType.class));
         player.registerObserver(this);
 
+        //initialize controls
+        HashMap<String, String> controlMap;
+
+        try {
+            controlMap = json.fromJson(HashMap.class, Gdx.files.local(PARTIAL_CONTROLS_SETTINGS_PATH));
```

Comment:
as you are loading the controls from a file you created in the same game session, the new controls are not activate unless you restart the game (that's probably why you added an if condition on the control settings page in game).

Reply:
Is not necessarily to restart application cause I've added the save of the json in the back button 👍

Reply:
indeed it saves the changes, but the controls are not applied in the current play session it seems.

Reply:
You have right, I'm sorry. I'll fix that asap :)